tests/utils/testing.rs

```diff
@@ -27,15 +29,172 @@ impl Drop for Cleanup {
+            "postgresql://postgres_user:test@localhost:5432/test_db".to_string(),
+            "postgres".to_string(),
+        )
+    } else {
+        (
+            format!("sqlite://tests/ayb_data_{slug}/ayb.sqlite"),
+            "sqlite".to_string(),
+        )
+    };
+
+    let e2e_testing_line = if test_type == "browser_sqlite" {
+        "e2e_testing = true\n"
+    } else {
+        ""
+    };
+
+    let config_content = format!(
+        r#"host = "0.0.0.0"
+port = {port}
+database_url = "{database_url}"
+data_path = "./tests/ayb_data_{slug}"
+{e2e_testing_line}
+[web]
+hosting_method = "Local"
+
+[email.file]
+path = "tests/ayb_data_{slug}/emails.jsonl"
+
+[authentication]
+fernet_key = "y3UdMqGh6si7pvQb8wsuW3ryiJcacp0H1QoHUPfsjb0="
+token_expiration_seconds = 3600
+
+[cors]
+origin = "*"
+
+[isolation]
+nsjail_path = "tests/nsjail"
+
+[snapshots]
+sqlite_method = "Vacuum"
+access_key_id = "minioadmin"
+secret_access_key = "minioadmin"
+bucket = "bucket"
+path_prefix = "{path_prefix}"
+endpoint_url = "http://localhost:9000"
+force_path_style = true
+
+[snapshots.automation]
+interval = "2s"
+max_snapshots = 6
+"#,
+        port = port,
+        database_url = database_url,
+        slug = slug,
+        e2e_testing_line = e2e_testing_line,
+        path_prefix = path_prefix
+    );
+
+    // Write the configuration to file
+    let mut file = std::fs::File::create(&config_path)?;
+    file.write_all(config_content.as_bytes())?;
+
+    Ok(config_path)
+}
+
+pub fn reset_test_environment(test_type: &str) -> Result<(), Box<dyn std::error::Error>> {
+    let data_dir = format!("./tests/ayb_data_{}", test_type);
+
+    match test_type {
+        "postgres" => {
+            // Remove data directory
+            if std::path::Path::new(&data_dir).exists() {
+                std::fs::remove_dir_all(&data_dir)?;
+            }
+
+            // Drop and recreate PostgreSQL database
+            let mut drop_cmd = Command::new("dropdb");
+            drop_cmd
+                .env("PGHOST", "localhost")
+                .env("PGUSER", "postgres_user")
+                .env("PGPASSWORD", "test")
+                .arg("test_db");
+
+            // Ignore error if database doesn't exist
+            let _ = drop_cmd.output();
+
+            let mut create_cmd = Command::new("createdb");
+            create_cmd
+                .env("PGHOST", "localhost")
+                .env("PGUSER", "postgres_user")
+                .env("PGPASSWORD", "test")
+                .arg("test_db");
+
+            let output = create_cmd.output()?;
+            if !output.status.success() {
+                return Err(format!(
+                    "Failed to create PostgreSQL database: {}",
+                    String::from_utf8_lossy(&output.stderr)
+                )
+                .into());
+            }
+        }
+        "sqlite" | "browser_sqlite" => {
+            // Remove data directory
+            if std::path::Path::new(&data_dir).exists() {
```

Comment:
Since all three tests do this, conditionally remove the data dir at the top of the function. It looks like we don't need a branch for sqlite/browser_sqlite at that point.